### PR TITLE
feat(web-components): read purses from bank instead of smart wallet

### PR DESCRIPTION
**Pull request by samsiegart**
refs https://github.com/Agoric/wallet-app/issues/49

For now the dapps only care about vbank assets so we can enhance this at a later point to include pure-js purses as well.

### CODE REVIEW

**Review by turadg — APPROVED** [review]
I imagine it's not practical right now, but what would it take to have a test that required updating for this change?

**Comment by samsiegart** [comment]
> I imagine it's not practical right now, but what would it take to have a test that required updating for this change?

I think for the effort to be worth the value we'd want e2e tests with a real local chain running. We could also just mock the queryclient or the follower to return the type of data we expect, but I don't know how much safer that makes us feel.

**Comment by dckc** [comment]
Mocking the query client would be great, I think.

File: packages/web-components/src/keplr-connection/queryBankBalances.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { QueryClient, createProtobufRpcClient } from '@cosmjs/stargate';
+import { Tendermint34Client } from '@cosmjs/tendermint-rpc';
+import { QueryClientImpl } from 'cosmjs-types/cosmos/bank/v1beta1/query';
+
+/**
+ * @param {string} address
+ * @param {string} rpc
+ */
+export const queryBankBalances = async (address, rpc) => {
```
Comment (dckc) [inline packages/web-components/src/keplr-connection/queryBankBalances.js:9]: another copy of this? same [scaling consideration](https://github.com/Agoric/wallet-app/pull/51#discussion_r1088314116) applies. Filed:

 - https://github.com/Agoric/agoric-sdk/issues/6870

File: packages/web-components/src/keplr-connection/queryBankBalances.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  const tendermint = await Tendermint34Client.connect(rpc);
+  const queryClient = new QueryClient(tendermint);
+  const rpcClient = createProtobufRpcClient(queryClient);
+  const bankQueryService = new QueryClientImpl(rpcClient);
+
+  const { balances } = await bankQueryService.AllBalances({
```
Comment (dckc) [inline packages/web-components/src/keplr-connection/queryBankBalances.js:15]: [ambient authority concern](https://github.com/Agoric/agoric-sdk/issues/2160) applies here too.

**Pull request merged**